### DISCUSSION TITLE
Using custom text editor with <Combobox />

### DISCUSSION THREAD

**Discussion by sndp-s** [ariakit/ariakit#2195] [Questions] [answered]
My use case is to render a menu on `@mention` in a ([doist/typist](https://github.com/Doist/typist)) rich text editor. 

I am using `<ComboboxPopover />`  to render my list, but I am not using the `<Combobox />` component as I am using another rich text editor.
```jsx
  <ComboboxPopover
    state={combobox}
    className="suggestion"
    portal
  >
    {items.map((member) => (
      <ComboboxItem
        className="suggestion__item"
        value={member.userId}
        key={member.userId}
      >
        <Text>{member.name}</Text>
      </ComboboxItem>
    ))}
  </ComboboxPopover>
  ```
  
I open the combobox this way.
  ```js
  React.useEffect(() => {
    combobox.show();
  });
  ```
  
  And I position the ComboboxPopover by supplying a custom `getAnchorRect()` to the `useComboboxState` hook
  ```js
    const combobox = useComboboxState({
      getAnchorRect: () => (decorationNode ? decorationNode.getBoundingClientRect() : null), // decorationNode is the @mention node
  });
  ```
  
  The problem I am facing is that `keyDown` ( up / down / escape ) events aren't navigating the ComboboxPopover list.

**Comment by diegohaz** [comment, marked as answer]
You have to render the underlying editor element (which I suppose has the `contenteditable` attribute) as a `<Combobox>` or pass the combobox component's props to it using render props. For example:

```jsx
<RichTextEditor
  contentEditable={
    <Combobox
      state={combobox}
      contentEditable
      autoSelect
      showOnKeyDown={false}
      showOnMouseDown={false}
      setValueOnChange={false}
      onPointerDown={combobox.hide}
      onKeyDown={(event) => {
        if (event.key === "ArrowLeft" || event.key === "ArrowRight") {
          combobox.hide();
        }
      }}
    />
  }
/>
```

We have a simple example using textarea that you can use as a reference: https://ariakit.org/examples/combobox-textarea

Also, there's this more complex example using Lexicaljs: https://ariakit-kmqava9a4-ariakit.vercel.app/examples/combobox-lexical
